lib/index.js

```diff
@@ -0,0 +1,103 @@
+/*!
+ * Copyright (c) 2024 Digital Bazaar, Inc. All rights reserved.
+ */
+// RFC 9285 The Base45 Data Encoding
+// https://www.rfc-editor.org/rfc/rfc9285
+
+// base45 alphabet (RFC 9285)
+const encodeTable = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ $%*+-./:';
+const decodeTable = new Uint8Array(256);
+
+// build decode table, using 0xFF as invalid value.
+decodeTable.fill(0xFF);
+for(let i = 0; i < encodeTable.length; ++i) {
+  decodeTable[encodeTable.charCodeAt(i)] = i;
+}
+
+// decode one char and check validity
+function _decodeChar(c) {
```

Comment:
Style nit: Put helper functions at the end, below publicly exposed API functions.